### ISSUE TITLE
Import detected unused, but in fact used

### ISSUE THREAD

**Issue by mredaelli** [PyCQA/pylint#3777]
### Steps to reproduce

Create the file `test.py`

```python
from math import e
from abc import ABC


class N(ABC):

    value: float

    def __init_subclass__(cls, v: float):
        cls.value = v


class E(N, v=e):
    pass
```

Run `pylint test.py`

### Current behavior

```
test.py:1:0: W0611: Unused e imported from math (unused-import)
```

### Expected behavior

The import is not unused.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.2 (default, Feb 25 2020, 10:39:28) 
[GCC 9.3.0]
```

**Comment by bmcgloin-tn** [comment]
I also get this. It's whenever the imported item is only used in the kwargs for a class definition.

**Comment by mbyrnepr2** [comment]
This one doesn't seem to be an issue anymore:

pylint --version

```
pylint 2.14.0-dev0
astroid 2.11.3
Python 3.10.0b2 (v3.10.0b2:317314165a, May 31 2021, 10:02:22) [Clang 12.0.5 (clang-1205.0.22.9)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you @mbyrnepr2, I get ``a.py:9:4: E0302: The special method '__init_subclass__' expects 0 param(s), 1 was given (unexpected-special-method-signature)`` on latest main.
